### PR TITLE
Enforce coverage

### PR DESCRIPTION
**Pull request by tomchristie**
Enforce test coverage when running the test suite.

Refs #316.

### CODE REVIEW

File: scripts/coverage [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+export PREFIX=""
+if [ -d 'venv' ] ; then
+    export PREFIX="venv/bin/"
+fi
+export SOURCE_FILES="httpx tests"
+
+set -x
+
+${PREFIX}coverage report --show-missing --skip-covered --fail-under=97
```
Comment (yeraydiazdiaz) [inline scripts/coverage:11]: I can't find `show-missing` on the coverage.py help or docs?

File: setup.cfg [hunk 2]
```diff
@@ -16,6 +16,6 @@ line_length = 88
 multi_line_output = 3
 
 [tool:pytest]
-addopts = --cov=httpx --cov=tests --cov-report=term-missing -rxXs
+addopts = --cov=httpx --cov=tests -rxXs
```
Comment (yeraydiazdiaz) [inline setup.cfg:19]: Should we include `--branch` as well?

Comment (StephenBrown2) [reply setup.cfg:19]: I am in favor of this.

File: scripts/coverage (hunk 1 above)
Comment (tomchristie) [reply scripts/coverage:11]: In the docs it’s only referred to by `-m`

Comment (yeraydiazdiaz) [reply scripts/coverage:11]: Ah, I see 👍

File: setup.cfg (hunk 2 above)
Comment (tomchristie) [reply setup.cfg:19]: I’m personally a pretty hard no on branch coverage.

Keeping things simple and pinning at 100% line coverage seems to absolutely be the sweet spot for code quality, and is a clear baseline that we can aim to enforce across all encode projects.

Comment (yeraydiazdiaz) [reply setup.cfg:19]: Personally I think if we are aiming for 100% we'd need branch coverage and rely on explicitly ignoring them in cases where it's not possible.

That being said, I'm not going to die on this particular hill 😅

**Pull request merged**